hollow/src/main/java/com/netflix/hollow/api/producer/HollowIncrementalProducer.java

```diff
@@ -69,6 +76,33 @@ public void delete(Object obj) {
     public void delete(RecordPrimaryKey key) {
         mutations.put(key, DELETE_RECORD);
     }
+
+    public static HollowIncrementalProducer.Builder withProducer(HollowProducer producer) {
+        HollowIncrementalProducer.Builder builder = new HollowIncrementalProducer.Builder();
+        return builder.withProducer(producer);
+    }
+    
+    public static class Builder {
```

Comment:
Do we need a builder just for one additional argument? Also, constructor for creating incremental producer is public. Just in my opinion it does not justify builder pattern. Consider removing it.

Reply:
I added the `Builder` pattern to keep consistence with the way `HollowProducer` and `HollowConsumer` are built. It's a fair point to say that it isn't enough to use the pattern. 

Regarding the constructor, I kept it public to avoid breaking existing consumers if they upgrade once this is merged. 

My question would be, Do you want to use the builder pattern and make the constructors private? or just have the constructors and remove the builder pattern?

Reply:
Just in case, changed the visibility to `protected` for the constructor

Reply:
I see why you added the `Builder` pattern. I think for now having a public constructor with `threadsPerCpu` will suffice. If in future we add more build parameters then we can add the `Builder` pattern. Keeping it simple.

Reply:
Makes sense @kineshsatiya 

I'll modify the code and let you know

Reply:
@kineshsatiya Applied your suggestion. Thanks for the feedback